packages/react-testing/src/matchers/components.ts

```diff
@@ -66,3 +66,46 @@ export function toContainReactComponent<
 
   return {pass, message};
 }
+
+export function toContainReactComponentTimes<
+  Type extends string | ComponentType<any>
+>(
+  this: jest.MatcherUtils,
+  node: Node<unknown>,
+  type: Type,
+  times: number,
+  props?: Partial<Props<Type>>,
+) {
+  assertIsNode(node, {
+    expectation: 'toContainReactComponentTimes',
+    isNot: this.isNot,
+  });
+
+  const foundByType = node.findAll(type);
+  const foundByProps =
+    props == null
+      ? foundByType
+      : foundByType.filter(element =>
+          Object.keys(props).every(key =>
+            this.equals(props[key], element.props[key]),
+          ),
+        );
+
+  const pass = foundByProps.length === times;
+
+  const message = pass
+    ? () =>
+        `${matcherHint('.not.toContainReactComponentTimes')}\n\n` +
+        `Expected the React element:\n  ${receivedColor(node.toString())}\n` +
+        `Not to contain component:\n  ${expectedColor(printType(type))}\n` +
+        `${times} ${pluralize('time', times)}, but it did.\n`
+    : () =>
+        `${matcherHint('.toContainReactComponentTimes')}\n\n` +
+        `Expected the React element:\n  ${receivedColor(node.toString())}\n` +
+        `To contain component:\n  ${expectedColor(printType(type))}\n` +
+        `${times} ${pluralize('time', times)}, but it was found ${
+          foundByProps.length
+        }.`;
```

Comment:
A little weird to end a sentence with a number.

Reply:
I'll add 'time(s)' at the end of the sentence.